packages/generator/src/sdk-metadata/code-sample-util.spec.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  VdmActionImport,
+  VdmFunctionImport,
+  VdmParameter,
+  VdmServiceMetadata
+} from '../vdm-types';
+import {
+  getActionFunctionImport,
+  getActionFunctionParams,
+  getFunctionWithMinParameters,
+  getFunctionWithoutParameters,
+  getLevensteinClosestFunction,
+  getODataEntity
+} from './code-sample-util';
+
+describe('code-sample-utils entity', () => {
+  it('gets entity based on levenshtein algorithm', () => {
+    const service = {
+      originalFileName: 'DummyClass',
+      entities: [{ className: 'DummyCollection' }, { className: 'DummyClass' }]
+    } as VdmServiceMetadata;
+    expect(getODataEntity(service.originalFileName, service.entities)).toEqual(
+      service.entities[1]
+    );
+  });
+
+  it('gets entity with shortest name when closest match is not found', () => {
+    const service = {
+      originalFileName: 'DummyClass',
+      entities: [{ className: 'TestEntity' }, { className: 'TestCollection' }]
+    } as VdmServiceMetadata;
+    expect(getODataEntity(service.originalFileName, service.entities)).toEqual(
+      service.entities[0]
+    );
+  });
+});
+describe('code-sample-utils imports', () => {
+  const functionImportsZeroParams = [
+    { name: 'serviceFunctionA', parameters: [] },
+    { name: 'serviceFunctionB', parameters: [{ parameterName: 'dummyParam' }] }
+  ] as VdmFunctionImport[];
+
+  const functionImportsMinParams = [
+    {
+      name: 'serviceFunctionA',
+      parameters: [{ parameterName: 'dummyParam1' }]
+    },
+    {
+      name: 'serviceFunctionB',
+      parameters: [
+        { parameterName: 'dummyParam2' },
+        { parameterName: 'dummyParam3' }
+      ]
+    }
+  ] as VdmFunctionImport[];
+
+  it('gets function import based on levenshtein algorithm', () => {
+    const functionImports = [
+      { name: 'some_Other_Function' },
+      { name: 'dummy_Class_Func' }
+    ] as VdmFunctionImport[];
+    expect(getLevensteinClosestFunction('DummyClass', functionImports)).toEqual(
+      functionImports[1]
+    );
+  });
+
+  it('gets undefined when func import based on levenshtein algorithm is not found', () => {
+    expect(
+      getLevensteinClosestFunction('DummyClass', functionImportsZeroParams)
+    ).toBeUndefined();
+  });
+
+  it('gets func import without parameters when no closest match found', () => {
+    expect(getFunctionWithoutParameters(functionImportsZeroParams)).toEqual(
```

Comment:
For both test this and the one below the "right" value is the first one. So you could swap the elements in the `functionImportsZeroParams` and assert that than the second element is returned as a second check here. Perhaps you implemente just give me the first :-).